### PR TITLE
Add shadcn library

### PR DESCRIPTION
**Pull request by dEdmishka**
I've added brand new React components library which includes all necessary components, like accordion, form, button e.c.

### CODE REVIEW

**Comment by DemianParkhomenko** [comment]
I think it is better to add components one by one and only when you need them. Also, components should be already modified according to our design.

**Comment by dEdmishka** [comment]
> I think it is better to add components one by one and only when you need them. Also, components should be already modified according to our design.

Yeap, let's add components when it'll be needed:)

File: src/components/Pages/Auth/FormAuth.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
 'use client'
 
-import Button from '@/components/UI/Button'
-import Form from '@/components/UI/Form/Form'
-import FormCheckbox from '@/components/UI/Form/FormCheckbox'
-import FormControl from '@/components/UI/Form/FormControl'
-import FormInput from '@/components/UI/Form/FormInput'
-import FormLabel from '@/components/UI/Form/FormLabel'
+import Button from '@/components/ui/index/Button'
```
Comment (firehawk89) [inline src/components/Pages/Auth/FormAuth.tsx:3]: Forgot to rename path

File: src/app/globals.css
Comment (firehawk89) [inline src/app/globals.css:1]: Maybe we should not use custom shadcn colors bcs we already have our design system

File: tailwind.config.ts
Comment (firehawk89) [inline tailwind.config.ts:1]: Here is the same about custom colors

File: src/app/globals.css
Comment (dEdmishka) [reply src/app/globals.css:1]: I think this colors defined by library in order to provide correct components functionality, but ok, I'll delete them.

File: tailwind.config.ts
Comment (dEdmishka) [reply tailwind.config.ts:1]: Yeah, removed

**Comment by DemianParkhomenko** [comment]
@dEdmishka Is it ready to review?

**Comment by dEdmishka** [comment]
> @dEdmishka Is it ready to review?

Em, it is.

**Comment by DemianParkhomenko** [comment]
Now build is failed. Please, fix it. You can run `npm run build` locally to check it.

image

**Comment by dEdmishka** [comment]
> Now build is failed. Please, fix it. You can run `npm run build` locally to check it.
> 
> image


build-example
Locally everything works great. I don't know, why your build failde. Try run npm install, or something.

**Review by DemianParkhomenko — CHANGES_REQUESTED** [review]
I think you should add a component from shadcn, to test how easily (difficult) you can use shadcn. Because now we cannot even check if our "theming" is right, and if does it fit for shadcn. Also, you can replace an existing component with a component from shadcn. 

And ideally the diff should not be very big. So then it will be easier to maintain our components. You can run `npx shadcn-ui diff` to check this. 

https://arc.net/l/quote/ntfpiwcp

**Comment by dEdmishka** [comment]
> I think you should add a component from shadcn, to test how easily (difficult) you can use shadcn. Because now we cannot even check if our "theming" is right, and if does it fit for shadcn. Also, you can replace an existing component with a component from shadcn. 
> 
> And ideally the diff should not be very big. So then it will be easier to maintain our components. You can run `npx shadcn-ui diff` to check this. 
> 
> https://arc.net/l/quote/ntfpiwcp

Did you find out why your build failed?

**Comment by DemianParkhomenko** [comment]
> > Now build is failed. Please, fix it. You can run `npm run build` locally to check it.

It fails in GitHub actions. But it fails locally too. 

image

I guess you renamed folder, but forgot to change tsconfig or something like that.

https://github.com/dev-KPI/messenger-frontend/blob/9b5c67b2cde153d0cfe4b19c74f69c7e11b53bba/tsconfig.json#L26

**Comment by dEdmishka** [comment]
> > > Now build is failed. Please, fix it. You can run `npm run build` locally to check it.
> 
> It fails in GitHub actions. Moreover it fails locally too. 
> 
> image
> 
> I guess you renamed folder, but forgot to change tsconfig or something like that.
> 
> https://github.com/dev-KPI/messenger-frontend/blob/9b5c67b2cde153d0cfe4b19c74f69c7e11b53bba/tsconfig.json#L26 

Yes, I changed the folder from incorrect 'UI' to 'ui', and lowercased some folders too, because some folders were lowercased, some not(I think, it's the right approach). So, I record these changes to config files too.

**Review by dEdmishka — COMMENTED** [review]
That's it. Returned library configs, tested library button in FormAuth.tsx. And I again checked configs, everything fine and dandy.
@DemianParkhomenko @firehawk89 pls renew your reviews

**Review by dEdmishka — COMMENTED** [review]
I've removed bugged alias "@/ui", 'cause in VS Code is simply ignoring it.

**Review by firehawk89 — APPROVED** [review]
LGTM

File: src/components/Pages/Auth/FormAuth.tsx [hunk 2]
```diff
@@ -28,6 +28,9 @@ const AuthForm: FC = () => {
       <Button className="mt-3 w-full" type="submit">
         Next
       </Button>
+      {/* <Button className="mt-3 w-full" type="submit">
+        Next
+      </Button> */}
```
Comment (firehawk89) [inline src/components/Pages/Auth/FormAuth.tsx:33]: You can remove this beacuse it's the same code as you've used above

Comment (dEdmishka) [reply src/components/Pages/Auth/FormAuth.tsx:33]: Ok

File: src/components/ui/index/Button.tsx
Comment (DemianParkhomenko) [inline src/components/ui/index/Button.tsx:1]: Let's remove this component. We already have a button from `shadcn`.  If it is blocked by `ButtonThemeToggle` @firehawk89 can make it in separate PR. 

You can use `Select` component from shadcn for this purpose or find different component which you like.

https://github.com/dev-KPI/messenger-frontend/blob/3594ee5517e406a7b25160605f8ddd8cb9b815d2/src/components/ui/index/ButtonThemeToggle.tsx#L8-L20

https://github.com/dev-KPI/messenger-frontend/assets/95881717/71014de1-f632-4feb-9433-106a19ba2fc3

https://ui.shadcn.com/docs/components/select

Comment (DemianParkhomenko) [inline src/components/ui/index/Button.tsx:1]: Please rename `src/components/ui/index` to `src/components/ui/custom` or something like that.

**Review by DemianParkhomenko — APPROVED** [review]
@dEdmishka Thanks for PR. Good job! Some minor comments to resolve and you can merge it.

**Comment by DemianParkhomenko** [comment]
@dEdmishka  @firehawk89 WDYT about renaming components files according to shadcn naming convention? 

i.e. to kebab case `ButtonThemeToggle.tsx` -> `button-theme-toggle.tsx`

File: src/components/ui/index/Button.tsx
Comment (DemianParkhomenko) [reply src/components/ui/index/Button.tsx:1]: Poll: https://github.com/dev-KPI/messenger-frontend/discussions/78

Comment (dEdmishka) [reply src/components/ui/index/Button.tsx:1]: There is no need to expand the _purpose_ of this mr(reminder - main purpose of this mr were to **install** and **mount** library shadcn to porject, **not rewrite\rename\delete** existed components\pages). Let's resolve sush problem in new issues and mrs. That's why I'm closing this conversation.

Comment (dEdmishka) [reply src/components/ui/index/Button.tsx:1]: Ok

Comment (dEdmishka) [reply src/components/ui/index/Button.tsx:1]: I don't understand the logic, Anton's components were simply removed to another folder, why not decide their fate in another mr's?

Comment (DemianParkhomenko) [reply src/components/ui/index/Button.tsx:1]: By custom, I mean components that are not in shadcn library.

Comment (DemianParkhomenko) [reply src/components/ui/index/Button.tsx:1]: >  Anton's components were simply removed to another folder, why not decide their fate in another mr's?

Ok, merge it :shipit:

Comment (DemianParkhomenko) [reply src/components/ui/index/Button.tsx:1]: > If it is blocked by ButtonThemeToggle @firehawk89 can make it in separate PR.

👆

**Pull request merged**